### PR TITLE
glass and sand fixes and code improvements (and chem dispensers)

### LINKED ISSUE #22258: You always slide on broken glass, never step on it
Being dragged over broken glass makes you slide on it, checks out

Walking over broken glass makes you slide on it, uhhhhh

### PR DESCRIPTION
**Pull request by Militaires**
- adds a cooldown to the sand-in-your-eyes messages.
- properly fixes the step-on-glass message.
- fixes you automatically re-wrenching the chem dispenser after you unwrench it.
- makes the sand-in-your-eyes message use [src] so that's that.

yeah sure

edit: dubs get.

edit: fixes #22258

### CODE REVIEW

**Comment by Cyberboss** [comment]
Tag the issues you are fixing

**Comment by PKPenguin321** [comment]
I'm not sure how this cool down is supposed to work considering it's only tied to one piece of sand and generally you're not throwing the same one over and over

File: code/modules/mining/ores_coins.dm [hunk 1]
```diff
@@ -90,18 +91,26 @@
 		return
 	var/mob/living/carbon/human/C = hit_atom
 	if(C.head && C.head.flags_cover & HEADCOVERSEYES)
-		visible_message("<span class='danger'>[C]'s headgear blocks the sand!</span>")
+		if(cooldown < world.time - 10)
```
Comment (RemieRichards) [inline code/modules/mining/ores_coins.dm]: why are you doing this backwards? this really isn't how we do these.

Comment (Militaires) [inline code/modules/mining/ores_coins.dm]: i dont get what you're saying

**Comment by Militaires** [comment]
@PKPenguin321 it's meant to stop the spam that happens when you're standing next to an explosion in lavaland and the amount of sand ash in your eyes is massive and annoying

what's the best way to do cds then

**Comment by optimumtact** [comment]
If you're fixing bugs, please add Fixes #relevant issue number to the PR body

File: code/modules/mining/ores_coins.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/mining/ores_coins.dm]: we usually do cooldown = world.time + {timetocooldown}

then check it as (if(cooldown > world.time)

**Comment by Militaires** [comment]
@optimumtact but i found both oranges myself when playing whatever ill make 2 new issue reports

**Comment by optimumtact** [comment]
People have already reported that glass bug, that was what I was referring to.

if it's an unreported one then don't bother.

**Comment by PKPenguin321** [comment]
I don't think this cooldown would actually work in practice. Test this

**Comment by Militaires** [comment]
@PKPenguin321 alright, ill use another method

File: code/modules/mining/ores_coins.dm [hunk 2]
```diff
@@ -90,18 +91,26 @@
...
+			visible_message("<span class='danger'>[C]'s headgear blocks the [src]!</span>")
+			cooldown = world.time
 		return
 	if(C.wear_mask && C.wear_mask.flags_cover & MASKCOVERSEYES)
-		visible_message("<span class='danger'>[C]'s mask blocks the sand!</span>")
-		return
+		if(cooldown < world.time - 10)
+			visible_message("<span class='danger'>[C]'s mask blocks the [src]!</span>")
+			cooldown = world.time
+			return
```
Comment (ChangelingRain) [inline code/modules/mining/ores_coins.dm]: this return is too indented

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 3]
```diff
@@ -167,7 +167,11 @@
 				. = TRUE
 
 /obj/machinery/chem_dispenser/attackby(obj/item/I, mob/user, params)
-	if(default_unfasten_wrench(user, I))
```
Comment (ChangelingRain) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: Uh, why are you changing this? If it's refastening, then you clicked it twice.

File: code/modules/mining/ores_coins.dm [hunk 4]
```diff
@@ -65,6 +65,7 @@
 	materials = list(MAT_GLASS=MINERAL_MATERIAL_AMOUNT)
 	refined_type = /obj/item/stack/sheet/glass
 	w_class = WEIGHT_CLASS_TINY
+	var/cooldown = 0
```
Comment (AnturK) [inline code/modules/mining/ores_coins.dm]: Did you miss some changes here ?

Comment (Militaires) [inline code/modules/mining/ores_coins.dm]: thanks

**Pull request closed without merging**